src/vs/workbench/parts/files/browser/views/explorerViewer.ts

```diff
@@ -111,6 +161,55 @@ export class FileDataSource implements IDataSource {
 		}
 	}
 
+	private resolveVirtualDirectories(files: FileStat[]) {
+		if (!this.enableVirtualDirectories) {
+			return;
+		}
+
+		files = files.filter(x => !x.isDirectory);
```

Comment:
Your logic is simpler, but it's method of 'choosing' a parent depends on the way `files` is sorted. I am going to try to simplify things while still getting the same results that I currently do.